sbnana/CAFAna/Systs/EnergySysts.h

```diff
@@ -0,0 +1,240 @@
+//ETW May 2020 Fresh version for SBN
+
+#pragma once
+
+#include "sbnana/CAFAna/Core/ISyst.h"
+#include "sbnana/CAFAna/Core/Utilities.h"
+#include "sbnana/CAFAna/Analysis/ExpInfo.h"
+#include "sbnanaobj/StandardRecord/Proxy/SRProxy.h"
+
+#include "TFile.h"
+#include "TH1.h"
+#include "TRandom3.h"
+
+#include <cassert>
+
+namespace ana
+{
+
+  enum class EnergyScaleSystTerm {
+    Constant,
+    Sqrt,
+    InverseSqrt
+  };
+
+  enum class EnergyScaleSystParticle {
+    Muon,
+    Hadron,
+    Neutron,
+    EM,
+    ChargedHadron
+  };
+
+  enum class EnergyScaleSystDetector {
+    SBND,
+    MicroBooNE,
+    ICARUS,
+    All
+  };
+
+  template<EnergyScaleSystTerm term, EnergyScaleSystParticle part, EnergyScaleSystDetector detector = EnergyScaleSystDetector::All>
+  class EnergyScaleSyst: public ISyst
+  {
+  public:
+    EnergyScaleSyst(double _uncertainty, std::string name) : 
+      ISyst(name, name), uncertainty(_uncertainty) {}
+    void Shift(double sigma, caf::SRSliceProxy *sr, double& weight) const override
+    {
+      double scale = uncertainty * sigma;
+      auto baseline_cut = [](double baseline){ 
+        auto all = (detector == EnergyScaleSystDetector::All);
+        auto nd = (detector == EnergyScaleSystDetector::SBND && baseline < 120);
+        auto ub = (detector == EnergyScaleSystDetector::MicroBooNE && baseline > 120 && baseline < 500);
+        auto fd = (detector == EnergyScaleSystDetector::ICARUS && baseline > 500);
+        return all || nd || ub || fd; 
+      };
+      if(sr->truth.iscc && abs(sr->truth.pdg) == 14 && !isnan(sr->fake_reco.nuE) && baseline_cut(sr->truth.baseline)) {
```

Comment:
I quite like early-return for this sort of thing. ie
```suggestion
      if(!sr->truth.iscc || abs(sr->truth.pdg) != 14 || isnan(sr->fake_reco.nuE) || !baseline_cut(sr->truth.baseline)) return;
```
and then the rest of the function can be one step less indented.